### DISCUSSION TITLE
How can I extract the EMA weights?

### DISCUSSION THREAD

**Discussion by mike4llison** [d8ahazard/sd_dreambooth_extension#348] [Q&A] [answered]
When trying to create the Model I get this message.
Checkpoint loaded from CheckpointInfo(filename='M:\\Stable\\Models\\v1-5-7GB.ckpt', title='v1-5-7GB.ckpt [a9263745]', hash='a9263745', model_name='v1-5-7GB', config='D:\\Github\\Automatic1111\\v1-inference.yaml')
Checkpoint M:\Stable\Models\v1-5-7GB.ckpt has both EMA and non-EMA weights.
In this conversion only the non-EMA weights are extracted. If you want to instead extract the EMA weights (usually better for inference), please make sure to add the `--extract_ema` flag.

Where can I add `--extract_ema` to get this to function? Inserting this into the webui_user.bat obviously doesn't work. Is this function included at all?

M.

**Comment by d8ahazard** [comment, marked as answer]
I've added a checkbox in the latest version of the UI that allows extracting the EMA weights. I'm interested to know how this works for training versus non-ema, as the text even says that non-ema is better for fine-tuning...which is why I originally didn't bother exposing it.

**Reply by mike4llison** [reply]
Wow, awesome. Tanks so much. I'll give it a go at the weekend.
